src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,455 @@
+        _setIssuanceToken(_issuanceToken);
+    }
+
+    //--------------------------------------------------------------------------
+    // Whitelist Management Functions
+
+    /// @notice Adds a single address to the whitelist
+    /// @param account_ The address to whitelist
+    function addToWhitelist(address account_) external onlyOrchestratorAdmin {
+        if (account_ == address(0)) revert FM_ExternalPrice__ZeroAddress();
+        
+        if (!_whitelist[account_]) {
+            _whitelist[account_] = true;
+            emit AddressWhitelisted(account_);
+        }
+    }
+
+    /// @notice Removes a single address from the whitelist
+    /// @param account_ The address to remove from whitelist
+    function removeFromWhitelist(address account_) external onlyOrchestratorAdmin {
+        if (_whitelist[account_]) {
+            _whitelist[account_] = false;
+            emit AddressRemovedFromWhitelist(account_);
+        }
+    }
+
+    /// @notice Adds multiple addresses to the whitelist
+    /// @param accounts_ Array of addresses to whitelist
+    function batchAddToWhitelist(
+        address[] calldata accounts_
+    ) external onlyOrchestratorAdmin {
+        uint256 length_ = accounts_.length;
+        if (length_ > MAX_BATCH_SIZE) revert FM_ExternalPrice__BatchSizeExceeded(length_);
+        
+        for (uint256 i_; i_ < length_; i_++) {
+            if (accounts_[i_] == address(0)) revert FM_ExternalPrice__ZeroAddress();
+            if (!_whitelist[accounts_[i_]]) {
+                _whitelist[accounts_[i_]] = true;
+                emit AddressWhitelisted(accounts_[i_]);
+            }
+        }
+        emit BatchWhitelistUpdated(accounts_, true);
+    }
+
+    /// @notice Removes multiple addresses from the whitelist
+    /// @param accounts_ Array of addresses to remove from whitelist
+    function batchRemoveFromWhitelist(
+        address[] calldata accounts_
+    ) external onlyOrchestratorAdmin {
+        uint256 length_ = accounts_.length;
+        if (length_ > MAX_BATCH_SIZE) revert FM_ExternalPrice__BatchSizeExceeded(length_);
+        
+        for (uint256 i_; i_ < length_; i_++) {
+            if (_whitelist[accounts_[i_]]) {
+                _whitelist[accounts_[i_]] = false;
+                emit AddressRemovedFromWhitelist(accounts_[i_]);
+            }
+        }
+        emit BatchWhitelistUpdated(accounts_, false);
+    }
+
+    /// @notice Checks if an address is whitelisted
+    /// @param account_ The address to check
+    /// @return isWhitelisted_ True if address is whitelisted
+    function isWhitelisted(
+        address account_
+    ) external view returns (bool isWhitelisted_) {
+        return _whitelist[account_];
+    }
+
+    //--------------------------------------------------------------------------
+    // View Functions
+
+    /// @param interfaceId_ The interface identifier to check support for
+    /// @return True if the interface is supported
+    function supportsInterface(
```

Comment:
Nope :P
It needs to check for the IFM_PC_ExternalPrice_Redeeming_v1 interface instead of IRedeemingBondingCurveBase_v1.
Because of the inheritance of the ERC20PaymentClientBase_v1 and RedeemingBondingCurveBase_v1 contracts you already check for their respective Interfaces, so IRedeemingBondingCurveBase_v1 for example is already done